pages/_app.js

```diff
@@ -1,10 +1,36 @@
 import GlobalStyle from "../styles";
+import { useState } from "react";
+import { initialShoppingList } from "@/lib/data";
+import { uid } from "uid";
+import sortShoppingListByCategory from "@/utils/sortShoppingListByCategory";
 
 export default function App({ Component, pageProps }) {
+  const [showForm, setShowForm] = useState(false);
+  const [currentTasks, setCurrentTasks] = useState(initialShoppingList);
+
+  function onCreateTask(taskData) {
+    const newTaskObject = { ...taskData, id: uid() };
+    console.log("New task:", newTaskObject);
+    setCurrentTasks((prevTasks) => [newTaskObject, ...prevTasks]);
+  }
+
+  const handleAddItem = (newItem) => {
+    setCurrentTasks([newItem, ...currentTasks]);
+  };
```

Comment:
I recommend combining the logic of those two function into this function:

```suggestion
  const handleAddItem = (newItem) => {
      const newTaskObject = { ...newItem, id: uid() };
    setCurrentTasks([newTaskObject, ...currentTasks]);
  };
```